src/iterators/idlist_iterator.c

```diff
@@ -142,58 +145,46 @@ static IteratorStatus MR_SkipTo(QueryIterator *base, t_docId docId) {
   IdListIterator *it = &mr->base;
   int rc = IL_SkipTo(base, docId);
   if (ITERATOR_OK == rc || ITERATOR_NOTFOUND == rc) {
-    base->current->num.value = mr->metricList[it->offset - 1];
-    SetYield(base);
+    SetYield(base, mr->metricList[it->offset - 1]);
   }
   return rc;
 }
 
 static void MR_Free(QueryIterator *self) {
   MetricIterator *mi = (MetricIterator *)self;
-  if (mi == NULL) {
-    return;
-  }
   IdListIterator *it = &mi->base;
   QueryIterator *base = &it->base;
-
   IndexResult_Free(base->current);
-
-  if (it->docIds) {
-    rm_free(it->docIds);
-  }
-
-  if (mi->metricList) {
-    rm_free(mi->metricList);
-  }
-
+  rm_free(it->docIds);
+  rm_free(mi->metricList);
   rm_free(mi);
 }
 
 QueryIterator *IT_V2(NewMetricIterator)(t_docId *docIds, double *metric_list, size_t num_results, Metric metric_type, bool yields_metric) {
```

Comment:
We said we should align the API here regarding array types (should be a simple allocation and not `array_t`, and also align on the ownership of the array (do we need to duplicate it or can we take ownership over it?)

Reply:
I thought it be easier not to change the API so that the transition from old iterators to new iterators is smoother.